float64 x
float64 y
float64 z
int16 time
---
int8 result
